apps/tenant-management-webapp-e2e/src/integration/calendar/calendar.step.ts

```diff
@@ -0,0 +1,205 @@
+import { Given, When, Then } from 'cypress-cucumber-preprocessor/steps';
+import commonlib from '../common/common-library';
+import CalendarPage from './calendar.page';
+
+const calendarObj = new CalendarPage();
+
+Given('a tenant admin user is on script service overview page', function () {
+  commonlib.tenantAdminDirectURLLogin(
+    Cypress.config().baseUrl,
+    Cypress.env('realm'),
+    Cypress.env('email'),
+    Cypress.env('password')
+  );
+  commonlib.tenantAdminMenuItem('Script', 4000);
+});
+
+Then('the user clicks Add calendar button on overview tab', function () {
+  calendarObj.addCalendarOverviewTabBtn().shadow().find('button').click({ force: true });
+  cy.wait(2000);
+});
+
+When('the user clicks Add calendar button', function () {
+  calendarObj.addCalendarBtn().shadow().find('button').click({ force: true });
+  cy.wait(2000);
+});
+
+Then('the user views Add calendar modal', function () {
+  calendarObj.addScriptModalTitle().invoke('text').should('eq', 'Add calendar');
+});
+
+When('the user clicks Save button in Add calendar modal', function () {
+  calendarObj.calendarModalSaveButton().shadow().find('button').click({ force: true });
+  cy.wait(1000);
+});
+
+When('the user clicks Cancel button in Add calendar modal', function () {
+  calendarObj.calendarModalCancelButton().shadow().find('button').click({ force: true });
+});
+
+Then('the user views Calendar tab table header', function () {
+calendarObj.calendarTableHeader().should('be.visible');
+});
+
+When('the user enters {string} in name field in calendar modal', function (name) {
+  calendarObj.addCalendarModalNameField().shadow().find('input').clear().type(name, { delay: 100, force: true });
+});
+
+Then('the user views the error message of {string} on namespace in calendar modal', function (errorMsg) {
+  calendarObj
+    .addCalendarModalNameFormItem()
+    .shadow()
+    .find('[class="error-msg"]')
+    .invoke('text')
+    .should('contain', errorMsg);
+});
+
+When('the user enters {string}, {string}, {string} in Add calendar modal',
+  function (name, desc, role) {
+    cy.viewport(1920, 1080);
+    calendarObj.addCalendarModalNameField().shadow().find('input').clear().type(name, { delay: 100, force: true });
+    calendarObj.addCalendarModalDescriptionField().shadow().find('.goa-textarea').clear().type(desc, { force: true });
+    // Select roles or client roles
+ const roles = role.split(',');
```

Comment:
Indentation seems odd. Do you have Prettier - Code formatter vs code extension installed? VS code should automatically add indentation for you.